x-pack/plugins/apm/public/components/app/Settings/agent_keys/index.tsx

```diff
@@ -129,58 +112,122 @@ function AgentKeysContent() {
+                defaultMessage: 'Create your first agent key',
+              })}
+            </h2>
+          }
+          body={
+            <p>
+              {i18n.translate('xpack.apm.settings.agentKeys.emptyPromptBody', {
+                defaultMessage:
+                  'Create agent keys to authorize requests to the APM Server.',
+              })}
+            </p>
+          }
+          actions={
+            <EuiButton
+              onClick={() => setIsFlyoutVisible(true)}
+              fill={true}
+              iconType="plusInCircleFilled"
+            >
+              {i18n.translate(
+                'xpack.apm.settings.agentKeys.createAgentKeyButton',
+                {
+                  defaultMessage: 'Create agent key',
+                }
+              )}
+            </EuiButton>
+          }
+        />
+      );
+    } else {
+      content = (
+        <AgentKeysTable
+          agentKeys={agentKeys ?? []}
+          refetchAgentKeys={refetchAgentKeys}
+        />
+      );
     }
   }
 
-  if (agentKeys && isEmpty(agentKeys)) {
-    return (
-      <EuiEmptyPrompt
-        iconType="gear"
-        title={
-          <h2>
-            {i18n.translate('xpack.apm.settings.agentKeys.emptyPromptTitle', {
-              defaultMessage: 'Create your first agent key',
-            })}
-          </h2>
-        }
-        body={
-          <p>
-            {i18n.translate('xpack.apm.settings.agentKeys.emptyPromptBody', {
-              defaultMessage:
-                'Create agent keys to authorize requests to the APM Server.',
-            })}
-          </p>
-        }
-        actions={
-          <EuiButton fill={true} iconType="plusInCircleFilled">
-            {i18n.translate(
-              'xpack.apm.settings.agentKeys.createAgentKeyButton',
-              {
-                defaultMessage: 'Create agent key',
-              }
-            )}
-          </EuiButton>
-        }
-      />
-    );
-  }
-
-  if (agentKeys && !isEmpty(agentKeys)) {
-    return (
-      <AgentKeysTable
-        agentKeys={agentKeys ?? []}
-        refetchAgentKeys={refetchAgentKeys}
-      />
-    );
-  }
-
-  return null;
+  return (
+    <Fragment>
+      <EuiText color="subdued">
+        {i18n.translate('xpack.apm.settings.agentKeys.descriptionText', {
+          defaultMessage:
+            'View and delete agent keys. An agent key sends requests on behalf of a user.',
+        })}
+      </EuiText>
+      <EuiSpacer size="m" />
+      <EuiFlexGroup justifyContent="spaceBetween">
+        <EuiFlexItem grow={false}>
+          <EuiTitle>
+            <h2>
+              {i18n.translate('xpack.apm.settings.agentKeys.title', {
+                defaultMessage: 'Agent keys',
+              })}
+            </h2>
+          </EuiTitle>
+        </EuiFlexItem>
+        {areApiKeysEnabled && canManage && (
+          <EuiFlexItem grow={false}>
+            <EuiButton
```

Comment:
I think this button should be hidden if the user is shown the empty state. We do this in the other settings views.
![CleanShot 2021-12-06 at 09 40 09@2x](https://user-images.githubusercontent.com/4104278/144838524-4291da39-6ba9-45f4-87c1-20a2284157d6.png)

Reply:
It makes sense, thanks. I've changed this so only the button in the prompt is visible in this case